### PR TITLE
Route/id to title

### PR DESCRIPTION
**Pull request by saadamir3180**
Change the ID-based routing to Title-based routing for Blog and Portfolio, also adjust the breadcrumbs for Blog and Portfolio. Added Placeholder data for Portfolio Projects and data/menu.js

### CODE REVIEW

File: data/menu.js [hunk 1]
```diff
@@ -64,9 +64,8 @@ export const menuItems = [
       {
         id: 31,
         label: "Penetration Testing & Vulnerability Assessment",
-        link: "/portfolio-vapt",
```
Comment (njmulsqb) [inline data/menu.js:67]: Why is this removed? The portfolio page is not opening now, it is directly sending the user to portfolio details page.

**Review by njmulsqb — CHANGES_REQUESTED** [review]
Remove ID-based routing from service details page as well, and anywhere else where this type of routing is used.

File: data/menu.js (hunk 1 above)
Comment (saadamir3180) [reply data/menu.js:67]: Actually, dynamic route is being used to display the portfolio projects instead of static routes that's why the new route with title is used.
For the portfolio page the View Our Work (in hero section) and Work With Us (in the header) button are being used.
Also the same flow was followed for the "pages" nav link which also has a subMenu.

Let me know your opinion on this.

File: data/menu.js [hunk 2]
```diff
@@ -64,9 +64,8 @@ export const menuItems = [
...
+        link: "/project-details/Penetration Test of Mach7t",
```
[the remark this answers is not in the document]
Comment (saadamir3180) [reply data/menu.js:67]: Noted.

File: data/features.js [hunk 3]
```diff
@@ -190,7 +190,7 @@ export const ourServices = [
   },
   {
     iconSrc: "/assets/img/icon/feature-icon1-3.svg",
-    title: "UI/UX Design",
```
Comment (njmulsqb) [inline data/features.js:193]: Why is this removed?

File: data/menu.js [hunk 4]
```diff
@@ -64,7 +64,7 @@ export const menuItems = [
       {
         id: 31,
         label: "Penetration Testing & Vulnerability Assessment",
-        link: "/project-details/Penetration Test of Mach7t",
+        link: "/project-details/Penetration-Test-of-Mach7t",
```
Comment (njmulsqb) [inline data/menu.js:67]: Why are we adding hyphens here when `replace` method is handing hyphens automatically?

Comment (njmulsqb) [reply data/menu.js:67]: image
Dashes should not show up in breadcrumbs

**Review by njmulsqb — CHANGES_REQUESTED** [review]

File: data/menu.js (hunk 1 above)
Comment (njmulsqb) [reply data/menu.js:67]: As "Penetration Testing & Vulnerability Assessment" is a broad category, clicking it should open its page that lists different portfolio items related to it, it should not directly route us to the Mach7t (one of the portfolio item) page.

Comment (saadamir3180) [reply data/menu.js:67]: Okay, got it. Additionally, there are more routes in the `app/(portfolio)` directory, such as `project-2`, `project-3`, etc. We need to fix those as well, but should I also add them to the `subMenu` of the `Portfolio` (NavLink)?

Comment (njmulsqb) [reply data/menu.js:67]: No, leave them as is. They're dummy, I'll add more data though

Comment (saadamir3180) [reply data/menu.js:67]: Okay, Thanks

File: data/features.js (hunk 3 above)
Comment (saadamir3180) [reply data/features.js:193]: Using `_` for `/` and `-` for ` ` (spaces) to encode and decode properly.

File: data/menu.js (hunk 4 above)
Comment (saadamir3180) [reply data/menu.js]: Done.

**Comment by njmulsqb** [comment]
image
Service details page is no longer working, it is returning 404.

**Comment by saadamir3180** [comment]
image

Working fine on my machine, going to push again

**Review by njmulsqb — APPROVED** [review]
